python_modules/dagster/dagster/_core/pipes/utils.py

```diff
@@ -268,20 +290,29 @@ def download_messages_chunk(self, index: int, params: PipesParams) -> Optional[s
     def _reader_thread(
         self, handler: "PipesMessageHandler", params: PipesParams, is_task_complete: Event
     ) -> None:
-        start_or_last_download = datetime.datetime.now()
-        while True:
-            now = datetime.datetime.now()
-            if (now - start_or_last_download).seconds > self.interval or is_task_complete.is_set():
-                chunk = self.download_messages_chunk(self.counter, params)
-                start_or_last_download = now
-                if chunk:
-                    for line in chunk.split("\n"):
-                        message = json.loads(line)
-                        handler.handle_message(message)
-                    self.counter += 1
-                elif is_task_complete.is_set():
-                    break
-            time.sleep(1)
+        try:
+            start_or_last_download = datetime.datetime.now()
+            while True:
+                now = datetime.datetime.now()
+                if (
+                    now - start_or_last_download
+                ).seconds > self.interval or is_task_complete.is_set():
+                    chunk = self.download_messages_chunk(self.counter, params)
+                    start_or_last_download = now
+                    if chunk:
+                        for line in chunk.split("\n"):
+                            message = json.loads(line)
+                            handler.handle_message(message)
+                        self.counter += 1
+                    elif is_task_complete.is_set():
+                        break
+                time.sleep(1)
+        except:
+            handler.report_component_exception(
+                f"{self.__class__.__name__} reader thread",
+                sys.exc_info(),
```

Comment:
Can we expand on error string here?

Exception thrown in the thread spawned by <<class nanme>> 

or something like that?

Reply:
[1] is the pre-amble